pkg/apis/seedmanagement/types_managedseedset.go

```diff
@@ -0,0 +1,128 @@
+	// Each ManagedSeed created / updated by the ManagedSeedSet will fulfill this template.
+	Template ManagedSeedTemplate
+	// ShootTemplate describes the Shoot that will be created if insufficient replicas are detected for hosting the corresponding ManagedSeed.
+	// Each Shoot created / updated by the ManagedSeedSet will fulfill this template.
+	ShootTemplate gardencore.ShootTemplate
+	// UpdateStrategy specifies the ManagedSeedSetUpdateStrategy that will be
+	// employed to update ManagedSeeds / Shoots in the ManagedSeedSet when a revision is made to
+	// Template / ShootTemplate.
+	UpdateStrategy *ManagedSeedSetUpdateStrategy
+	// RevisionHistoryLimit is the maximum number of revisions that will
+	// be maintained in the ManagedSeedSet's revision history. Defaults to 10.
+	RevisionHistoryLimit *int32
+}
+
+// ManagedSeedSetUpdateStrategy specifies the strategy that the ManagedSeedSet
+// controller will use to perform updates. It includes any additional parameters
+// necessary to perform the update for the indicated strategy.
+type ManagedSeedSetUpdateStrategy struct {
+	// Type indicates the type of the ManagedSeedSetUpdateStrategy. Defaults to ManagedSeedSetUpdateStrategyType.
+	Type *ManagedSeedSetUpdateStrategyType
+	// RollingUpdate is used to communicate parameters when Type is ManagedSeedSetUpdateStrategyType.
+	RollingUpdate *RollingUpdateManagedSeedSetUpdateStrategy
+}
+
+// ManagedSeedSetUpdateStrategyType is a string enumeration type that enumerates
+// all possible update strategies for the ManagedSeedSet controller.
+type ManagedSeedSetUpdateStrategyType string
+
+const (
+	// RollingUpdateManagedSeedSetUpdateStrategyType indicates that update will be
+	// applied to all ManagedSeeds / Shoots in the ManagedSeedSet with respect to the ManagedSeedSet
+	// ordering constraints.
+	RollingUpdateManagedSeedSetUpdateStrategyType ManagedSeedSetUpdateStrategyType = "RollingUpdate"
+)
+
+// RollingUpdateManagedSeedSetStrategy is used to communicate parameter for RollingUpdateManagedSeedSetUpdateStrategyType.
+type RollingUpdateManagedSeedSetUpdateStrategy struct {
+	// Partition indicates the ordinal at which the ManagedSeedSet should be partitioned. Defaults to 0.
+	Partition *int32
+}
+
+// ManagedSeedSetStatus represents the current state of a ManagedSeedSet.
+type ManagedSeedSetStatus struct {
+	// ObservedGeneration is the most recent generation observed for this ManagedSeedSet. It corresponds to the
+	// ManagedSeedSet's generation, which is updated on mutation by the API Server.
+	ObservedGeneration int64
+	// Replicas is the number of replicas (ManagedSeeds and their corresponding Shoots) created by the ManagedSeedSet controller.
+	Replicas int32
+	// ReadyReplicas is the number of ManagedSeeds created by the ManagedSeedSet controller that have a Ready Condition.
+	ReadyReplicas int32
+	// NextReplicaNumber is the ordinal number that will be assigned to the next replica of the ManagedSeedSet.
+	NextReplicaNumber int32
```

Comment:
This is needed for the "ever increasing sequence" of replica ordinal numbers to work. We need to store somewhere the next replica ordinal number (starting with 0), and the status of the resource is the most obvious choice. When a new replica is created, it will get this number as ordinal, and the value in the status will be incremented by 1. The StatefulSet doesn't need such a field since its ordinals are calculated differently.
Come think of it, because of this difference, there might be other fields needed in the status, but it's rather hard to foresee all of them now. I will add them when the need for them becomes more clear.

Reply:
okay thanks for the explanation.